nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -776,6 +797,58 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
         }
     }
 
+    private void setParameter(PreparedStatement ps, int index, Object value, int fieldSqlType, int sqlType) throws IOException {
+        if (sqlType == Types.BLOB) {
+            // Convert Byte[] into Blob
+            if (fieldSqlType == Types.ARRAY) {
+                if (!(value instanceof byte[])) {
+                    if (value == null) {
+                        try {
+                            ps.setNull(index, Types.BLOB);
+                            return;
+                        } catch (SQLException e) {
+                            throw new IOException("Unable to setNull() on prepared statement" , e);
+                        }
+                    } else {
+                        throw new IOException("Expected BLOB to be of type byte[] but is instead " + value.getClass().getName());
+                    }
+                }
+                byte[] byteArray = (byte[]) value;
+                try (InputStream inputStream = new ByteArrayInputStream(byteArray)) {
+                    ps.setBlob(index, inputStream);
+                } catch (SQLException e) {
+                    throw new IOException("Unable to parse binary data " + value, e.getCause());
+                }
+            } else {
+                try (InputStream inputStream = new ByteArrayInputStream(value.toString().getBytes())) {
```

Comment:
Should the specific character set be specified with the call to `getBytes()` or is it intentional to have this call be dependent on the runtime system properties?  Is it possible to add a unit test to exercise this block?

Reply:
This is another one of those cases like the above, a corner case I don't even know exists. Maybe if the target column is a BLOB but the field is a String or something, basically any time the final "else" clause above is reached, where we just convert the data according to the field and column types without the special BLOB/CLOB handling. I can try some more unit tests to exercise this.

Reply:
Thanks for making the updates, just noticed this line should be adjusted to explicitly specify UTF-8 for `getBytes()` as opposed to relying on platform defaults.

Reply:
Oops missed that one, will update now